js/video/software.js

```diff
@@ -1,1606 +1,1657 @@
-	}
-	this.scalerot = new Array(32);
-	for (var i = 0; i < 32; ++i) {
-		this.scalerot[i] = {
-			a: 1,
-			b: 0,
-			c: 0,
-			d: 1
-		};
+	load8(offset) {
+		return (this.loadU8(offset) << 24) >> 24;
 	}
-};
-
-GameBoyAdvanceOAM.prototype = Object.create(MemoryAligned16.prototype);
-
-GameBoyAdvanceOAM.prototype.overwrite = function(memory) {
-	for (var i = 0; i < (this.buffer.byteLength >> 1); ++i) {
-		this.store16(i << 1, memory[i]);
+	load16(offset) {
+		return (this.loadU16(offset) << 16) >> 16;
 	}
-};
-
-GameBoyAdvanceOAM.prototype.store16 = function(offset, value) {
-	var index = (offset & 0x3F8) >> 3;
-	var obj = this.objs[index];
-	var scalerot = this.scalerot[index >> 2];
-	var layer = obj.priority;
-	var disable = obj.disable;
-	var y = obj.y;
-	switch (offset & 0x00000006) {
-	case 0:
-		// Attribute 0
-		obj.y = value & 0x00FF;
-		var wasScalerot = obj.scalerot;
-		obj.scalerot = value & 0x0100;
-		if (obj.scalerot) {
-			obj.scalerotOam = this.scalerot[obj.scalerotParam];
-			obj.doublesize = !!(value & 0x0200);
-			obj.disable = 0;
-			obj.hflip = 0;
-			obj.vflip = 0;
-		} else {
-			obj.doublesize = false;
-			obj.disable = value & 0x0200;
-			if (wasScalerot) {
-				obj.hflip = obj.scalerotParam & 0x0008;
-				obj.vflip = obj.scalerotParam & 0x0010;
-			}
+	loadU8(offset) {
+		var index = offset >> 1;
+		if (offset & 1) {
+			return (this.buffer[index] & 0xFF00) >>> 8;
+		}
+		else {
+			return this.buffer[index] & 0x00FF;
 		}
-		obj.mode = (value & 0x0C00) >> 6; // This lines up with the stencil format
-		obj.mosaic = value & 0x1000;
-		obj.multipalette = value & 0x2000;
-		obj.shape = (value & 0xC000) >> 14;
-
-		obj.recalcSize();
-		break;
-	case 2:
-		// Attribute 1
-		obj.x = value & 0x01FF;
-		if (obj.scalerot) {
-			obj.scalerotParam = (value & 0x3E00) >> 9;
-			obj.scalerotOam = this.scalerot[obj.scalerotParam];
-			obj.hflip = 0;
-			obj.vflip = 0;
-			obj.drawScanline = obj.drawScanlineAffine;
-		} else {
-			obj.hflip = value & 0x1000;
-			obj.vflip = value & 0x2000;
-			obj.drawScanline = obj.drawScanlineNormal;
-		}
-		obj.size = (value & 0xC000) >> 14;
-
-		obj.recalcSize();
-		break;
-	case 4:
-		// Attribute 2
-		obj.tileBase = value & 0x03FF;
-		obj.priority = (value & 0x0C00) >> 10;
-		obj.palette = (value & 0xF000) >> 8; // This is shifted up 4 to make pushPixel faster
-		break;
-	case 6:
-		// Scaling/rotation parameter
-		switch (index & 0x3) {
-		case 0:
-			scalerot.a = (value << 16) / 0x1000000;
-			break;
-		case 1:
-			scalerot.b = (value << 16) / 0x1000000;
-			break;
-		case 2:
-			scalerot.c = (value << 16) / 0x1000000;
-			break;
-		case 3:
-			scalerot.d = (value << 16) / 0x1000000;
-			break;
-		}
-		break;
-	}
-
-	MemoryAligned16.prototype.store16.call(this, offset, value);
-};
-
-function GameBoyAdvancePalette() {
-	this.colors = [ new Array(0x100), new Array(0x100) ];
-	this.adjustedColors = [ new Array(0x100), new Array(0x100) ];
-	this.passthroughColors = [
-		this.colors[0], // BG0
```

Comment:
Why where these comments removed? I think they are helpful

Reply:
the refactoring to ES6 was done automatically by VSCode which must have automatically removed the comments, I can re-add them.

Reply:
Ok no problem. That's odd that it removed some but not others. Actually give me a few minutes I'm about to make a PR to your branch where I'll re-add them